helmfile/charts/networkpolicy-workload/templates/ingress-nginx.yaml

```
@@ -0,0 +1,98 @@
+{{ if .Values.ingressNginx.enabled }}
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: ingress-nginx-deny-all
+  namespace: ingress-nginx
+spec:
+  podSelector: {}
+  policyTypes:
+    - Ingress
+    - Egress
+
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: ingress-nginx-np
+  namespace: ingress-nginx
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/component: controller
+  policyTypes:
+    - Ingress
+    - Egress
+  ingress:
+    - ports:
+      - port: 443
+      - port: 80
+    - from:
+        - namespaceSelector:
+            matchExpressions:
+              - key: kubernetes.io/metadata.name
+                operator: In
+                values:
+                  - monitoring
+        - podSelector:
+            matchLabels:
+              app.kubernetes.io/instance: kube-prometheus-stack-prometheus                    
+      ports:
+        - port: 10254     
+    - from:
+{{- range $IP := .Values.global.apiserver.ips }}
+      - ipBlock:
+          cidr: {{ $IP }}
+{{- end }}
+      ports:
+        - protocol: TCP
+          port: 8443           
+  egress:
+    - ports:
+      - port: 53
+        protocol: UDP
+      - port: 443
+      - port: 80
+    - to:
+        {{- range $IP := .Values.global.apiserver.ips }}
+        - ipBlock:
+            cidr: {{ $IP }}
+        {{- end }}
+      ports:
+        - protocol: TCP
+          port: {{ .Values.global.apiserver.port }} 
+    - to:
+        - namespaceSelector:
+            matchExpressions:
+              - key: kubernetes.io/metadata.name
+                operator: In
+                values:
+                  - kube-system
+        - podSelector:
+            matchLabels:
+              component: kube-apiserver                  
+      ports:
+        - port: 3000   
+    - to:
+        - namespaceSelector: {}
+
+---
+
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: ingress-nginx-deny-all
```

Maybe rename this to
```suggestion
  name: ingress-nginx-default-backend
```

---

Do you want me to ? 

I think deny all says a bit about what it does.

---

But this is for default backend. You have another deny-all.

---

Changed